client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -46,6 +53,20 @@ object PathBlob {
     }
   }
 
+  /** Factory method to combine two PathBlob instances
+   *  @param firstBlob: first pathblob
+   *  @param secondBlob: second pathblob
+   *  @return failure if the two blobs have conflicting entries; the combined blob otherwise.
+   */
+  def reduce(firstBlob: PathBlob, secondBlob: PathBlob): Try[PathBlob] = {
+    if (firstBlob.conflicts(secondBlob)) {
```

Comment:
This should have tipped you off that the function name is wrong, since you want to fail on conflict.

Reply:
Yeah you're right. I should have looked at this more closely.